pkg/compact/compact.go

```diff
@@ -496,6 +497,7 @@ func (cg *Group) compact(ctx context.Context, dir string, comp tsdb.Compactor) (
 	}
 	if len(plan) == 0 {
 		// Nothing to do.
+		level.Debug(cg.logger).Log("msg", "Nothing to compact in this compaction group.", "group", cg.Key())
```

Comment:
Hm this is quite spammy. Do we actually want this? Is there any case that we don't get log (info/debug) when `plan > 0`? AFAIK we always does.

Reply:
Number of groups: num of scrapers x 3 (downsampling)
For me it is ~10 scrapers in single environment so 30 debug logs. Additionally,on initial deploys I migrated scrapers from some old to the new name, then I modified again, so I had some super old blocks (7 weeks ago?) that will log every 5 minutes, but we are sure no compaction should be in place.

And maybe that number still fits into debug log verbocity, but I just don't think it is that useful - especially for old blocks.

I have an idea. Can we make some metric instead and increase some useful counter (even per group) that will give you needed info? I think this is a good place to use metric (number) vs log line. What do you think?

Reply:
There are already metrics to indicate when compaction occurred versus compaction attempts; I think more detailed metrics would inappropriate. Part of the benefit of the debug log entries is that they include the line number and file generating the entry, which helps when debugging issues.

I'll remove this commit for now since it's not required for this PR.